coffee/base.py

```diff
@@ -969,25 +969,26 @@ class FunDecl(Statement):
 
     Syntax: ::
 
-        [pred] ret name ([args]) {body};
+        [template] [pred] ret name ([args]) {body};
 
     E.g.: ::
 
         static inline void foo(int a, int b) {return;};"""
 
-    def __init__(self, ret, name, args, body, pred=[], headers=None):
+    def __init__(self, ret, name, args, body, pred=[], headers=None, template=[]):
```

Comment:
Instead of setting the default value of function arguments a reference to a mutable value (a list in this case), their default value should be `None`, and then you initialise the instance attribute to `pred or []`. This will initialise `self.pred` to a new empty list, rather than the same one.